codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/parse/JsonParserGenerator.kt

```diff
@@ -374,6 +378,22 @@ class JsonParserGenerator(
         }
     }
 
+    private fun RustWriter.deserializeBigInteger() {
+        rustTemplate(
+            "#{expect_string_or_null}(tokens.next())?.map(|s| s.to_unescaped().map(|u| #{BigInteger}::new(u.into_owned()))).transpose()?",
```

Comment:
Also many (all?) of the supported JSON protocols represent these as regular JSON numbers: https://smithy.io/2.0/aws/protocols/aws-json-1_0-protocol.html#shape-serialization